src/arrayMethodSort.js

```diff
@@ -4,8 +4,26 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const conditionFunction =
+      compareFunction ||
+      function (a, b) {
+        return String(a) > String(b) ? 1 : -1;
+      };
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        const a = this[i];
+        const b = this[j];
+        const result = conditionFunction(a, b);
+
+        if (result > 0) {
+          [this[i], this[j]] = [this[j], this[i]];
+        }
+      }
+    }
```

Comment:
You are using a bubble sort algorithm, which is fine for this task. However, please note that bubble sort is not the most efficient sorting algorithm for larger arrays. For future tasks, you might want to consider using more efficient algorithms like quicksort or mergesort.